From 593d162ffdf33af7dc242c913957afe752d61b38 Mon Sep 17 00:00:00 2001
From: "River, Li" <river.li@intel.com>
Date: Tue, 26 Apr 2016 16:39:00 +0800
Subject: [PATCH 1/4] SQUASHME! [VPG]: drm/i915: Fix IPANIC of
 intel_get_crtc_scanline.

When connect HDMI, __intel_set_mode will call intel_update_maxfifo()
to disable maxfifo, in which __intel_get_crtc_scanline() is called
to read scanline, then IPANIC since crtc->config.adjusted_mode
is not ready.

We can fix it as Sivakumar suggestion:
Before accessing the mode struct check if crtc is active.
If not active we should return 0.

This patch has to be squashed into
Already Verified: https://jira01.devtools.intel.com/browse/OAM-26200
Change-Id: Iad99c4fb5051d651ef2bd4913ba8e3fa4f535a66
Signed-off-by: Shen, Lei <lei.shen@intel.com>
Reviewed-on: https://android.intel.com:443/477648
FOR_UPSTREAM [VPG]: drm/i915: Enhance ftrace for atomic update

Change-Id: I460f67b9058e24124b108a01bd6d92eab693c9a3
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30345
Signed-off-by: River, Li <river.li@intel.com>
Reviewed-on: https://android.intel.com:443/499841
---
 drivers/gpu/drm/i915/i915_irq.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 08468c3..920c925 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -26,6 +26,7 @@
  *
  */
 
+
 #define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
 
 #include <linux/sysrq.h>
@@ -928,10 +929,14 @@ static int __intel_get_crtc_scanline(struct intel_crtc *crtc)
 {
 	struct drm_device *dev = crtc->base.dev;
 	struct drm_i915_private *dev_priv = dev->dev_private;
-	const struct drm_display_mode *mode = &crtc->config.adjusted_mode;
+	const struct drm_display_mode *mode;
 	enum pipe pipe = crtc->pipe;
 	int position, vtotal;
 
+	if (!crtc->active)
+		return 0;
+
+	mode = &crtc->config.adjusted_mode;
 	vtotal = mode->crtc_vtotal;
 	if (mode->flags & DRM_MODE_FLAG_INTERLACE)
 		vtotal /= 2;
-- 
1.9.1

